### PR TITLE
I737 log prior means

### LINKED ISSUE #737: Create mean functions for each log prior
This makes it easier to check what parameterisation is implemented.

### PR DESCRIPTION
**Pull request by fcooper8472**
Fixes #737 

This follows on from #736 which ins't merged yet, so not sure how this works in terms of a PR...

Looks as though we see all changes from master, so probably worth merging the other PR first?

### CODE REVIEW

**Comment by MichaelClerx** [comment]
> This follows on from #736 which ins't merged yet, so not sure how this works in terms of a PR...

> Looks as though we see all changes from master, so probably worth merging the other PR first?

Thanks @fcooper8472 . #736 is now merged into master, but the PR didn't change immediately.

I wasn't sure how to get github to notice the difference, so I edited your PR twice, setting the base to <not master> and then to master again, and it seems to have picked up that it only needs 3 commits now!

File: pints/_log_pdfs.py [hunk 1]
```diff
@@ -40,6 +40,13 @@ def evaluateS1(self, x):
         """
         raise NotImplementedError
 
+    def mean(self):
```
Comment (MichaelClerx) [inline pints/_log_pdfs.py]: I think this should be in `LogPrior`, not in `LogPDF`. In general we don't know the mean of the LogPDFs, that's the whole point :-)

File: pints/_log_priors.py [hunk 2]
```diff
@@ -76,6 +76,10 @@ def evaluateS1(self, x):
             return value, np.asarray([np.divide(self._a - 1., _x) - np.divide(
                 self._b - 1., 1. - _x)])
 
+    def mean(self):
+        """ See :meth:`LogPDF.mean()`. """
```
Comment (MichaelClerx) [inline pints/_log_priors.py]: See comment above

File: pints/_log_priors.py [hunk 3]
```diff
@@ -255,6 +263,10 @@ def evaluateS1(self, x):
         else:
             return value, np.asarray([-self._rate])
 
+    def mean(self):
+        """ See :meth:`LogPDF.mean()`. """
+        return 1. / self._rate
```
Comment (MichaelClerx) [inline pints/_log_priors.py]: We don't need the `1.` here, it can just be `1`

File: pints/_log_priors.py [hunk 4]
```diff
@@ -786,6 +826,13 @@ def evaluateS1(self, x):
         # much...
         return self(x), np.zeros(self._n_parameters)
 
+    def mean(self):
+        """ See :meth:`LogPDF.mean()`. """
+        if isinstance(self._boundaries, pints.RectangularBoundaries):
```
Comment (MichaelClerx) [inline pints/_log_priors.py]: nice!

File: pints/_log_pdfs.py (hunk 1 above)
Comment (MichaelClerx) [reply pints/_log_pdfs.py]: Also, we don't really expect most/many LogPDF classes to implement this method; having it on the interface makes it look too much like a requirement for me

**Comment by ben18785** [comment]
#744 rebases this and so closing

**Pull request closed without merging**